src/HandlerParser/parser_test.go

```diff
@@ -1,13 +1,43 @@
 package handlerparser
 
 import (
-	"gingen/src"
+	"bufio"
+	"os"
+	"regexp"
 	"testing"
-
 )
 
+var comment_regexp = regexp.MustCompile(`^[ |\t]*//.*`)
+
+/** @brief This function is used to get the file content and return it as an array of string with one line for each case.
+ * @param path The path to the file. The path MUST BE VALID.
+ * @param only_comment If true, the function will return only the comments. If false, the function will return the whole file.
+ * @return []string the content of the file or an error if the file cannot be read.
+ */
+func readFile(path string, only_comment bool) ([]string, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	// Close the file when the function is done
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	scanner.Split(bufio.ScanLines)
+	var text []string
+	// Read the file line by line and add it to the text slice
+	for scanner.Scan() {
+		line := scanner.Text()
+		if only_comment && comment_regexp.MatchString(line) {
+			text = append(text, line)
+		} else if !only_comment {
+			text = append(text, line)
+		}
+	}
+	return text, nil
+}
```

Comment:
looks like this function is shared between tests, you can maybe put it in a dedicated file 😄

Reply:
It is already in a separate file (fileHandler.go) but within test if i include parent module it does something weird and no let me import the module